docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,174 @@
+# Contribution Guide
+
+1. [Prerequisites](#prerequisites)
+2. [Build](#build)
+3. [Check](#check)
+4. [Testing](#testing)
+5. [Pull-request](#pull-request)
+6. [License](#license)
+
+Thank you for your time and effort to help us improve samba-operator.
+Here are a few steps to get started, with reference to some useful
+resources.
+
+
+## Prerequisites
+
+Development effort takes place using Linux environment and requires at
+minimum:
+
+1. [Go 1.18](https://golang.org/dl/) installed
+2. [GitHub](https://github.com/) account
+3. Development tools: git, make, and podman or docker
+4. Testing: [minikube](https://minikube.sigs.k8s.io)
+5. [Quay](https://quay.io/) account (optional)
+
+Development collaboration takes place via github's samba-in-kubernetes
+([SINK](https://github.com/samba-in-kubernetes/)) pages and facilities.
+
+The top-level [Makefile](../Makefile) is the entry point for various
+build commands. Few utilities are required during the build process,
+either for bootstrapping the project, building it or checking the
+validity of the source code itself. If not present in your `$PATH`,
+those tools may be installed locally to the project (under the
+local `.bin` directory) using:
+
+```sh
+  $ make build-tools
+  ...
+  $ ls .bin/
+```
+
+## Build
+
+The following section describes how to build and test samba-operator.
+A more detailed description of this process may be found in the
+[developer-notes](./developer-notes.md) document.
+
+Building the samba-operator is a straightforward action with
+`make build`. Upon successful build, the output binary is written to
+`bin/manager`:
+
+```sh
+  $ make build
+  ...
+  $ ls bin/
+  manager
+```
+
+Note that the manager binary alone is not typically executed in a
+Kubernetes cluster. For that we generally build a container image. To
+build an OCI container image including the manager binary, run:
+
+```sh
+  $ make image-build
+```
+
+The name of this image is controlled by the `IMG` and `TAG` variables
+and may be customized to fit your needs. A developer may override those
+as well as some other default Makefile settings via the `devel.mk` file
+at the project's root directory:
+
+```sh
+  $ cp devel.mk.sample devel.mk
+  $ vim devel.mk
+```
+
+## Check
+
+Before submitting any pull request, a developer must ensure that
+his code satisfies the following minimum requirements:
+
+1. Go code conforms to the project's standards.
+2. YAML files conforms to the project's standards.
```

Comment:
Unfortunately, this is a gramar regression. `YAML files conform` is correct  (or `YAML file conforms`).

Reply:
If you are confused as to why the line above is "conforms", and is also correct, that's because "Go code" is treated as a singular. (As a collective of plural things like files, lines of code, etc. A bit like how "the ocean" is made of "water molecules" ;-)   I think there's an exception in UK English for companies, but I don't think that applies here.)

Reply:
ACK